backend/src/main/java/com/stock/admin/controller/ProductController.java

```diff
@@ -0,0 +1,121 @@
+    @ResponseBody
+    public ProductsResponse getAllProductsOfASpecificShop(@PathVariable("shopId") String shopCode) {
+        List<Product> products = productService.getAllProductsByShopId(shopCode);
+        ProductsResponse productsResponse = new ProductsResponse();
+        productsResponse.setStatus(Response.ResponseStatus.SUCCESS);
+        productsResponse.setPayload(products);
+        return productsResponse;
+    }
+
+    /**
+     * Gets products details.
+     *
+     * @param productCode the product code
+     * @return the products details
+     */
+    @RequestMapping(value = "/{productCode}", method = RequestMethod.GET, produces = "application/json; charset=UTF-8")
+    @ResponseBody
+    public ProductResponse getProductsDetails(@PathVariable("productCode") String productCode) {
+        Product product = productService.getByProductId(productCode);
+        ProductResponse productResponse = new ProductResponse();
+        productResponse.setStatus(Response.ResponseStatus.SUCCESS);
+        productResponse.setPayload(product);
+        return productResponse;
+    }
+
+
+    /**
+     * Add product product response.
+     *
+     * @param productRequest the create product request
+     * @return the product response
+     */
+    @RequestMapping(method = RequestMethod.POST, produces = "application/json; charset=UTF-8")
+    @ResponseBody
+    public ProductResponse addProduct(@RequestBody ProductRequest productRequest) {
+        Product addedProduct = productService.create(productRequest);
+        ProductResponse productResponseBody = new ProductResponse();
+        productResponseBody.setPayload(addedProduct);
+        productResponseBody.setStatus(Response.ResponseStatus.SUCCESS);
+        return productResponseBody;
+    }
+
+
+    /**
+     * Modify by product by id product response.
+     *
+     * @param productId      the product id
+     * @param productRequest the product request
+     * @return the product response
+     */
+    @RequestMapping(value = "/{id}", method = RequestMethod.PUT)
+    @ResponseBody
+    public ProductResponse modifyByProductById(@PathVariable("id") String productId, @Valid @RequestBody ProductRequest productRequest) {
+        Product updatedProduct = productService.updateByProductId(productId, productRequest.getProduct());
+        ProductResponse productResponseBody = new ProductResponse();
+        productResponseBody.setPayload(updatedProduct);
+        productResponseBody.setStatus(Response.ResponseStatus.SUCCESS);
+        return productResponseBody;
+    }
+
+
+    /**
+     * Delete by product by id.
+     *
+     * @param productId      the product id
+     * @param productRequest the product request
+     */
+    @RequestMapping(value = "/{id}", method = RequestMethod.DELETE)
+    @ResponseBody
+    public void deleteByProductById(@PathVariable("id") String productId, @Valid @RequestBody ProductRequest productRequest) {
```

Comment:
You don't have the body for product request. And it is not required also

Reply:
I agree. The changed one is already committed and I was expecting it to be part of this PR.  Looking for the cause.